src/scripting/scriptingService.ts

```diff
@@ -43,23 +51,62 @@ export class ScriptingService {
         7: 'SqlServerStretchEdition'
     };
 
+    /**
+     * Helper to node name from label and metadata
+     */
+    private getObjectName(nameLabel: string, objectName: string, metadata: ObjectMetadata[]): string {
+        if (nameLabel === objectName) {
+            return nameLabel;
+        }
+        // if the names aren't the same, find the closest
+        // name to the name label
+        let closestName = '';
+        for (const obj of metadata) {
+            if (nameLabel.includes(obj.name)) {
+                if (obj.name.length > closestName.length) {
+                    closestName = obj.name;
+                }
+            }
+        }
+        return closestName;
+    }
+
     /**
      * Helper to get the object name and schema name
-     * @param node
      */
-    private getObjectNames(node: TreeNodeInfo): string[] {
-        let fullName = node.label;
-        let objects = fullName.split('.');
-        return objects;
+    private async getObjectFromNode(node: TreeNodeInfo, uri: string): Promise<ScriptingObject> {
+        const nodeCredentials = node.connectionCredentials;
+        let metadata: ObjectMetadata[];
+        for (let credential of this._credentialsToMetadataMap.keys()) {
```

Comment:
nit: This is a bit simpler

`const metadata = Array.from(this._credentialsToMetadataMap.keys()).find(c => Utils.isSameConnection(c, nodeCredentials));`